context/llpcCompiler.cpp

```diff
@@ -823,6 +832,170 @@ Result Compiler::BuildShaderModule(
     return result;
 }
 
+Result Compiler::BuildPipelineUsingRelocatableElf(
+    Context*                            pContext,                   // [in] Acquired context
+    ArrayRef<const PipelineShaderInfo*> shaderInfo,                 // [in] Shader info of this pipeline
+    uint32_t                            forceLoopUnrollCount,       // [in] Force loop unroll count (0 means disable)
+    ElfPackage*                         pPipelineElf)               // [out] Output Elf package
+{
+    Result result = Result::Success;
+
+    // Determine whether or not relocatable elf can be used.
+    // We only do relocatable elf if it is a VsPs shader that is given in
+    // spir-v.
+    bool useRelocatableShaders = true;
+    for (uint32_t stage = 0; stage < shaderInfo.size(); ++stage)
+    {
+        if (stage != ShaderStageVertex && stage != ShaderStageFragment)
+        {
+            if ((shaderInfo[stage] != nullptr) && (shaderInfo[stage]->pModuleData != nullptr))
+            {
+                useRelocatableShaders = false;
+            }
+        }
+        else if (shaderInfo[stage] == nullptr || shaderInfo[stage]->pModuleData == nullptr)
```

Comment:
Yes, I agree.  I guess some late changes I made did not make it through.  I've added a TODO here.  I don't want to handle it now because I need to figure out how to generate a pass through fragment shader as relocatable shader elf that can be cached and linked.  It will not be handled on the first pass.